### PR TITLE
Run YCbCr channel upsampling per-group.

### PR DESCRIPTION
**Pull request by veluca93**
No difference in pixel values (at least to the 6th digit). This
significantly reduces memory usage for the subsampled cases.

```
Before:
Read 181018 compressed bytes [v0.3.7 | SIMD supported: AVX2,SSE4,Scalar]
Decoded to pixels.
../lib/extras/codec_pnm.cc:372: PNM encoder ignoring metadata - use a different codec
Done.
2268 x 1512, 76.23 MP/s [76.23, 76.23], 1 reps, 32 threads.
Allocations: 593 (max bytes in use: 2.235271E+08)
../lib/extras/codec_pnm.cc:292: PNM: no color_space/icc_pathname given, assuming sRGB
1.2382063866
3-norm: 0.372505

After:
Read 181018 compressed bytes [v0.3.7 | SIMD supported: AVX2,SSE4,Scalar]
Decoded to pixels.
../lib/extras/codec_pnm.cc:372: PNM encoder ignoring metadata - use a different codec
Done.
2268 x 1512, 116.55 MP/s [116.55, 116.55], 1 reps, 32 threads.
Allocations: 717 (max bytes in use: 2.081835E+08)
../lib/extras/codec_pnm.cc:292: PNM: no color_space/icc_pathname given, assuming sRGB
1.2382063866
3-norm: 0.372505
```

### CODE REVIEW

**Review by lvandeve — APPROVED** [review]
LGTM

**Comment by lvandeve** [comment]
Approved, once the failing test is fixed

File: lib/jxl/dec_reconstruct.cc [hunk 1]
```diff
@@ -287,6 +287,114 @@ void FloatToRGBA8(const Image3F& input, const Rect& input_rect, bool is_rgba,
...
+                       const Rect& frame_rect, const FrameDimensions& frame_dim,
+                       ImageF* plane_out, const LoopFilter& lf, ImageF* temp) {
+  // The pixel in (xoff, yoff) is the origin of the downsampling coordinate
+  // system.
+  size_t xoff = PassesDecoderState::kGroupDataXBorder;
+  size_t yoff = PassesDecoderState::kGroupDataYBorder;
+  // This may over-copy, but it should always be safe to do so.
+  size_t y0 = rect.y0() - lf.Padding();
+  size_t y1 = rect.y0() + rect.ysize() + lf.Padding();
+  size_t x0 = rect.x0() - lf.Padding();
```
Comment (lvandeve) [inline lib/jxl/dec_reconstruct.cc:300]: Is it possible for lf.Padding() to be > rect.xo()?

Comment (lvandeve) [reply lib/jxl/dec_reconstruct.cc:300]: Actually I mean whether it can be >= rect.x0() given the x0 - 1 below

Comment (veluca93) [reply lib/jxl/dec_reconstruct.cc:300]: it shouldn't be, IIRC `rect.x0() >= RoundUpTo(lf.Padding(), 8) + 8`

**Review by lvandeve — APPROVED** [review]

**Pull request merged**